### ISSUE TITLE
Category Autocomplete doesn't seem to work when Flat Categories are activated

### ISSUE THREAD

**Issue by fcapovilla** [Smile-SA/elasticsuite#147]
On our stores, the "Use Flat Catalog Category" preference is activated.
After updating to Elasticsuite 2.2.1 to use the Category Autocomplete feature, we have encountered errors when triggering the autocomplete : 

```
2016/09/28 17:07:16 [error] 535#0: *7873 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Call to undefined method Magento\Catalog\Model\ResourceModel\Category\Flat::getAttributeRawValue() in /www/sites/.../public/vendor/smile/elasticsuite/src/module-elasticsuite-catalog/Model/Autocomplete/Category/ItemFactory.php:220
```

Our current configuration is :

Elasticsuite 2.2.1
Magento 2.1.1
Elasticsearch 2.3.5

**Comment by romainruaud** [comment]
Hello @fcapovilla and thank you for submitting an issue. 

We'll provide a fix to this one soon.

Regards,

**Comment by afoucret** [comment]
This one have been fixed by merging PR #150

You can use it now by using `master`, `2.2.x`or `2.3.x`.
It will be part of the next releases : 
- 2.2.2 that will provide fixes to 2.2.1
- 2.3.0 that is our next big release)

Thx for reporting
